met-web/src/components/engagement/view/PreviewBanner.tsx

```diff
@@ -10,50 +10,43 @@ export const PreviewBanner = () => {
     const navigate = useNavigate();
     const { engagementLoading, savedEngagement, publishEngagement } = useContext(ActionContext);
     const isLoggedIn = useAppSelector((state) => state.user.authentication.authenticated);
+    const [isDraft, setIsDraft] = useState(true);
 
     const engagementId = savedEngagement.id || '';
-    if (engagementLoading) {
-        return <Skeleton variant="rectangular" width="100%" height="10em" />;
-    }
-    const isDraft = savedEngagement.status_id === EngagementStatus.Draft;
 
-    const validate = () => {
-        const errors = {
-            status: savedEngagement.status_id !== EngagementStatus.Draft,
-        };
-
-        return Object.values(errors).some((isError: unknown) => isError);
-    };
+    useEffect(() => {
+        setIsDraft(savedEngagement.status_id === EngagementStatus.Draft);
```

Comment:
I had to use useEffect for it to work

Reply:
I see, weird thought because we are already using this pattern somewhere else and it's working. Like const isNewEngagement = engagementId === 'create';  in EngagementForm. The setIsDraft needs to be in a useEffect for sure. But a simple const would still work no? Would you mind trying again just using const isDraft = savedEngagement.status_id === EngagementStatus.Draft?

Reply:
It worked!